### ISSUE TITLE
Fix package name does not match import path

### ISSUE THREAD

**Issue by breerly** [yarpc/yarpc-go#135]
Right now we have something like:

```
import "github.com/yarpc/yarpc-go"

rpc := yarpc.New()
```

Which feels wrong because `yarpc` doesnt match the package name, `yarpc-go`.

I suggest moving all go code into a package `yarpc` like so:

```
import "github.com/yarpc/yarpc-go/yarpc"

rpc := yarpc.New()
```

The import paths aren't the prettiest, but I think it's a good tradeoff.

@yarpc/golang

**Comment by breerly** [comment]
The other option would be to give Go special treatment in our orgs, and let it have the `yarpc/yarpc` repo name. We'll have to find another home for the top-level docs, but that's an option we should consider.

**Comment by blampe** [comment]
Not worth doing right now.

**Comment by breerly** [comment]
I disagree - now is the best time to make a change like this.

The cost will be nearly 0 now, but expensive if we wait till this is released.

**Comment by breerly** [comment]
Per discussion with @abhinav @prashantv - here's the options.

## Use subpackage

This means moving code into a subpackage `yarpc`, it fixes the "package name not matching import path" problem, but creates quite the stutter:

```
import "github.com/yarpc/yarpc-go/yarpc"

rpc := yarpc.New()
```

## Use canonical import paths

This allows our code to stay the same, but gives us "vanity" import paths:

https://golang.org/doc/go1.4#canonicalimports

This is exactly how gRPC allows importing from `google.golang.org/grpc` instead of `github.com/grpc/grpc-go`:

https://godoc.org/google.golang.org/grpc

Since we own `yarpc.io`, we can do something like:

```
import "go.yarpc.io/yarpc"

rpc := yarpc.New()
```

or
 
```
import "go.uber.com/yarpc"

rpc := yarpc.New()
```

**Comment by abhinav** [comment]
I like the second approach better. The subpackage approach has a lot of stuttering. Calling it `go.yarpc.io/yarpc` or `go.uber.com/yarpc` sounds much better.

**Comment by prashantv** [comment]
+1 to canonical import paths. It also means we have more freedom in our repo layout.

**Comment by breerly** [comment]
`go.uber.com` is not going to be an option, that's currently being used for driver signup and I wasn't able to inspire OSS to take it over :)
 
The following is probably the best we can do:

```
import "go.yarpc.io/yarpc"

rpc := yarpc.New()
```

**Comment by breerly** [comment]
I'll check one more time for `golang.uber.com` - these look pretty nice this way:

```
import (
    golang.uber.com/yarpc
    golang.uber.com/crossdock
    golang.uber.com/thriftrw
)

...
```

**Comment by breerly** [comment]
`go.uber.org` looks even better:

```
import (
    go.uber.org/yarpc
    go.uber.org/crossdock
    go.uber.org/thriftrw
    go.uber.org/openctx
    go.uber.org/jaeger
)

...
```

Checking with OSS.

**Comment by breerly** [comment]
Fixed by #294

**Comment by breerly** [comment]
Reopening since this was reverted in `0.1.1` - Let's get this queued up for `0.2.0` instead.

**Comment by breerly** [comment]
This is now solved in `~0.3`.
